docs/extend/authorization.md

```diff
@@ -0,0 +1,243 @@
+authentication plugin(s). The request contains the user (caller) and command
+context. The plugin is responsible for deciding whether to allow or deny the
+request.
+
+Below please find tow sequence diagrams describing the *allow* and *deny* 
+authorization flows:
+
+![Bad Address](images/authz_allow.png)
+
+![Bad Address](images/authz_deny.png)
+
+Each request sent to the plugin includes the authenticated user, the HTTP
+headers, and the request/response body. Only the user name and the
+authentication method used are passed to the plugin. Most importantly, no user
+credentials or tokens are passed. Finally, not all request/response bodies
+are sent to the authorization plugin. Only those request/response bodies where
+the `Content-Type` is either `text/*` or `application/json` are sent.
+
+For commands that can potentially the hijack the HTTP connection (`HTTP
+Upgrade`), such as `exec`, the authorization plugin are only called for the
+initial HTTP requests. Once the plugin approves the command, authorization is
+not applied to the rest of the flow. Specifically, the streaming data is not
+passed to the authorization plugins. For commands that return chunked HTTP
+response, such as `logs` and `events`, only the HTTP request are sent to the
+authorization plugins as well.
+
+During request/response processing, some authorization plugins flows might
+need to do additional queries to the Docker daemon. To complete such flows,
+plugins can call the daemon API similar to a regular user. To enable these
+additional queries, the plugin must provide the means for an administrator to
+configure proper authentication and security policies.
+
+## Docker client flows
+
+To enable and configure the authorization plugin, the plugin developer must support the Docker client interactions detailed in this section.
+
+### Setting up Docker daemon
+
+Enable the authorization plugin with a dedicated command line flag in the
+`--authz-plugins=PLUGIN_ID` format. The flag supplies a `PLUGIN_ID` value.
+This value can be the plugin’s socket or a path to a specification file.
+
+```
+$ docker daemon --authz-plugins=plugin1 --auth-plugins=plugin2,...
+```
+
+Docker's authorization subsystem supports multiple `--authz-plugin` parameters.
+
+### Calling authorized command (allow)
+
+Your plugin must support calling the `allow` command to authorize a command. This call does not impact Docker's command line.
+
+```
+$ docker pull centos
+...
+f1b10cd84249: Pull complete
+...
+```
+
+### Calling unauthorized command (deny)
+
+Your plugin must support calling the `deny` command to report on the outcome of a plugin interaction. This call returns messages to Docker's command line informing the user of the outcome of each call.  
+
+```
+$ docker pull centos
+…
+Authorization failed. Pull command for user 'john_doe' is denied by authorization plugin 'ACME' with message ‘[ACME] User 'john_doe' is not allowed to perform the pull command’
```

Comment:
Can you wrap this example output as well? The code blocks in the documentation aren't very wide, and people will have to scroll to see the whole text otherwise